src/OpenLoco/src/Audio/Audio.cpp

```diff
@@ -785,23 +788,24 @@ namespace OpenLoco::Audio
         }
     }
 
-    static void off_4FEB58(Vehicles::VehicleSoundPlayer* v, int32_t x)
+    // 0x004FEB58
+    static void processVehicleForSound(Vehicles::VehicleSoundPlayer* v, int32_t step)
     {
-        switch (x)
+        switch (step)
         {
             case 0:
+                // 0x048A262
                 v->soundFlags &= ~Vehicles::SoundFlags::flag0;
                 break;
             case 1:
-                if ((v->soundFlags & Vehicles::SoundFlags::flag1) == Vehicles::SoundFlags::none)
-                {
-                    sub_48A274(v);
-                }
+                // 0x048A268
+                triggerVehicleSoundIfInView(v);
```

Comment:
This `flag1` check was not present in the original game for `case 1`.

Reply:
Yes it is 
<img width="565" height="176" alt="image" src="https://github.com/user-attachments/assets/df8cbfc9-938e-4c33-8724-3d448647c3e7" />